File: front_end/src/app/Services/bordereau.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { Observable } from 'rxjs';
import { Bordereaux } from '../Models/Bordereaux';

@Injectable({
  providedIn: 'root'
})
export class BordereauService {

  private baseUrl = 'http://localhost:8034/api/bordereaux' ;

  constructor(private http: HttpClient) { }

  genererBordereau(prestationsIds: number[]): Observable<any> {
    return this.http.post<any>(`${this.baseUrl}/genererBorderau`, prestationsIds)
  }

  getAllBordereaux(): Observable<Bordereaux[]> {
    return this.http.get<Bordereaux[]>(`${this.baseUrl}/findAll`);
  }

  getBordereauxByUserId(userId: number): Observable<Bordereaux[]> {
    return this.http.get<Bordereaux[]>(`${this.baseUrl}/getBordereauxByUserId/${userId}`);
  }

  getBordereauxById(id: number): Observable<Bordereaux> {
    return this.http.get<Bordereaux>(`${this.baseUrl}/getByID/${id}`);
  }

  deleteBordereau(id: number): Observable<void> {
    return this.http.delete<void>(`${this.baseUrl}/${id}`);
  }

  changerStatutBordereau(id: number, nouveauStatut: string): Observable<void> {
    return this.http.put<void>(`${this.baseUrl}/${id}/changerStatut?nouveauStatut=${nouveauStatut}`, {});
  }





}

File: backend/src/main/java/com/example/projectPfe/services/Interface/BordereauService.java
package com.example.projectPfe.Services.Interface;

import com.example.projectPfe.models.Bordereaux;
import com.example.projectPfe.models.EstatusBord;

import java.util.List;

public interface BordereauService {

    public Bordereaux genererBordereau(List<Integer> prestationsIds);

    public Bordereaux findById(int id);

    public List<Bordereaux> findAll();

    public List<Bordereaux> getBordereauxByUserId(int userId) ;

    public void deleteBordereauById(int bordereauId);

    public void changerStatutBordereau(int id, EstatusBord nouveauStatut) ;
}
